advocacy_docs/pg_extensions/wait_states/index.mdx

```diff
@@ -0,0 +1,17 @@
+---
+title: EDB Wait States
+directoryDefaults:
+  product: EDB Wait States
+
+---
+
+EDB Wait States is a tool for analyzing performance and tuning. The EDB Wait States contrib module contains two main components: the EDB Wait States Background Worker (EWSBW) and the EDB Wait States interface. 
```

Comment:
- Can we provide any more detailed information on the general use of EDB Wait States? I lifted the basic description from the EPAS docs. 
- Are the two components EWSBW and the interface? Originally, the docs didn't mention a second beside EWSBW. 
- Should we provide any additional info here on the interface/other component? All of the info below is about EWSBW and more info on the interface is in Using.

Reply:
EDB Wait States is a tool for analysing performance and tuning by allowing the collection and querying of Wait Event data. Wait Events, introduced in PostgreSQL 9.6, are recorded alongside other session activity and provide a snap shot of whether a session is waiting for I/O, CPU, IPC, locks or timeouts. Snapshots of this information are gathered by the EDB Wait States background worker (EWSBW) at regular intervals. The EDB Wait States interface allows you to control when and for how long the wait events are sampled, and to extract the gathered samples (in `edb_wait_states_data` for further analysis. By gathering this data over time, optimization opportunities can be discovered and insight gained into what resources sessions are waiting on when performance is lower than expected.

Reply:
Correct. Basically EDB wait states, collect the wait event data which gets generated in the backend (server) and create a snapshot.  Can refer to the PG document for the type of wait event types and wait events.  (https://www.postgresql.org/docs/current/monitoring-stats.html#WAIT-EVENT-ACTIVITY-TABLE)